src/cascadia/TerminalApp/Profile.h

```diff
@@ -93,9 +98,10 @@ class TerminalApp::Profile final
     static winrt::Microsoft::Terminal::Settings::CursorStyle _ParseCursorShape(const std::wstring& cursorShapeString);
     static std::wstring_view _SerializeCursorStyle(const winrt::Microsoft::Terminal::Settings::CursorStyle cursorShape);
 
-    static GUID _GenerateGuidForProfile(const std::wstring& name) noexcept;
+    static GUID _GenerateGuidForProfile(const std::wstring& name, const std::optional<std::wstring>& source) noexcept;

```

Comment:
o*line discussion around using `const json::Value&` instead of unpacking it.

Reply:
I actually split it up into two parts - one for unpacking the json (`GetGuidOrGenerateForJson`), and another for doing the source/name GUID generation. I had to split it up like that because the profile doesn't actually have a reference to its own JSON. i could have theoretically had the profile re-create a blob of json and call the json parsing function, but that seemed insane.